oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,428 @@
+  administrators`_ that they can no longer continue in the role
+* A maintainer has been unreachable for more than 30 days
+* A maintainer has violated the project `Code of Conduct`_
+* A maintainer hasn't met project quality standards consistently
+  despite feedback
+
+In the case that a maintainer has violated the `Code of Conduct`_, the
+Goverance Team, specified in the Code of Conduct, will decide what the
+repercussions of the violation will be, including whether the violator
+can continue in the role of maintainer.  It should be noted that
+maintainers are expected to be exemplars and, as such, will be held to
+a high standard.
+
+In the case that a maintainer is not meeting project quality standards
+consistently, the scrum-of-scrums will inform the maintainer with the
+expectation that their components meet standards within a calendar
+month.  If, after a month, the quality issues are not resolved, the
+maintainer will be removed by the scrum-of-scrums.  Additionally, a
+maintainer who is warned three (3) times in a single calendar year
+will also be removed. The `program administrators`_ serve as a
+backstop - any community member may contact them at any time if
+a problem arises and the normal channels of communication are
+not working.
+
+Documenting Maintainers
+-----------------------
+As it is essential that maintainers are discoverable and reachable,
+the person or persons responsible for maintaining a component should
+be listed in the repository README file along with the mechanism for
+contacting a maintainer.
+
+Jobs of the Maintainer
+----------------------
+Many of the day-to-day specifics of the maintainer role are outside of
+the scope of this document.  For example, here we will note that
+maintainers are responsible for ensuring that repositories meet
+project standards, but not the details of those standards.  Those
+standards - including establishing a channel for feedback, and details on how
+and when that channel will be monitored - will be codified in an OEP over time,
+but will initially be defined in the wiki page `requirements for public
+repositories`_ to allow rapid iteration.
+
+The key aspects of the maintainer role fall into the following
+categories
+
+* **Community Stewardship**: Maintainers are representatives of the
+  project in the community and are responsible for contributing to an
+  environment that is welcoming, vibrant, and growing.
+* **Project Management**: Maintainers are responsible for ensuring
+  that critical maintenance is done, but this doesn't mean doing it all
+  themselves. Effective delegation is an important maintainer skill.
+* **Quality Assurance**: Maintainers are responsible for the quality
+  of their components.  Quality will mean meeting defined project
+  standards.  It can also mean improving our standards or improving
+  our processes.
+* **Technical Vision**: Maintainers should have a documented technical
```

Comment:
I think that that should be captured as well.  When we think about a README for any component, we'd typically want to know why the thing exists and how it is engineered.  I believe that both can be captured very succinctly. The technical vision might be a little more specific on details.

Reply:
@adamstankiewicz this is another place where I think the word "technical" may be a hang-up for your team, whereas I don't think your team should consider yourself un-technical. Are you able to provide a vision (technical or otherwise) that answers the questions that follow?

```
  * What is the component for?
  * How it fits into the overall platform?
  * What are its limitations and key areas of debt?
  * What new patterns or technologies should be adopted to improve the
    component?
  * How it can be extended
```

Is there a rewording or something missing here that you'd like to see?

Reply:
We definitely wouldn't consider the Paragon team/component un-technical, but wanted to surface that Paragon goes beyond a technical vision alone. For example, the design vision for Paragon and how we do design across the organization helps inform/lead some aspects of our technical vision for Paragon (and vice versa), likely why I was getting hung up on explicitly calling it a technical vision alone (i.e., the vision for components like Paragon are all-encompassing of both a design and technical vision that coincide together).

The technical vision for Paragon, as a result, then would actually be more of a "here's where we're going with design and, as such, here's what we need to do technically", for example.

If I were to suggest a change in the wording, it'd likely be referring to a "Component Vision" versus a "Technical Vision" specifically. But, don't consider this as blocking feedback; I don't feel strongly about the word choice as either way, the technical vision for Paragon will include aspects of the design vision in order to account for the aforementioned questions.

Reply:
@adamstankiewicz I actually think that's a good point - I would absolutely want a component's vision to be comprised not just of future technical architecture work but where the component is going (ie, what's on its product roadmap). In the interest of getting this resolved quickly I'll touch base with Ed on this point - would you think a "technical + product vision" wording encompass your thoughts?

Reply:
Ok @adamstankiewicz Ed & I have come to an agreement on our thoughts here, being that, eventually we would love to see Maintainers document both a technical and a design/product vision. The current state of most components is that latter vision is not well formed/well available, and we're currently working on some of that via the Product Roadmap and Product Working Group.

Ultimately, we're going to choose to leave the wording as-is, but we would LOVE if components (especially ones like Paragon) started moving towards the direction of capturing the entire component vision. And after we roll this out, I think we'll be ready to do an iteration on this OEP that does bring the product/design vision more closely into the fold.

Reply:
This makes sense to me, the goal here is not to define everything a maintainer is *allowed* to do but rather the things they *must* do.  For components like paragon, that are ahead of the curve, I think doing more and showing the value of having product vision is a great way to lead by example to push our standards even further in the future.